domains/client/relayer/src/worker.rs

```diff
@@ -363,63 +174,44 @@ where
     // then fetch new messages in the block
     // construct proof of each message to be relayed
     // submit XDM as unsigned extrinsic.
-    while let Some(block) = chain_block_imported.next().await {
+    while let Some(imported_block) = chain_block_imported.next().await {
         // if the client is in major sync, wait until sync is complete
         if sync_oracle.is_major_syncing() {
             tracing::debug!(target: LOG_TARGET, "Client is in major sync. Skipping...");
             continue;
         }
 
-        if !block.is_new_best {
+        if !imported_block.is_new_best {
             tracing::debug!(target: LOG_TARGET, "Imported non-best block. Skipping...");
             continue;
         }
 
-        let (number, hash) = {
-            let imported_block_number = *block.header.number();
-            if let Some(block_to_relay) = imported_block_number.checked_sub(&confirmation_depth_k) {
-                let block_to_relay_hash = consensus_client
-                    .hash(block_to_relay)?
-                    .ok_or(Error::MissingBlockHash)?;
-                (block_to_relay, block_to_relay_hash)
-            } else {
-                tracing::debug!(target: LOG_TARGET, "Not enough confirmed blocks. Skipping...");
-                continue;
-            }
+        let Some(finalized_block_number) = imported_block
```

Comment:
Please use a different name, maybe `confirmed_block_number`. This block is not finalized from Substrate point of view and can lead to confusion when reading the code.